compiler/modules/types/string.bal

```diff
@@ -1,65 +1,152 @@
 // Implementation specific to basic type string.
 
 type StringSubtype readonly & record {|
+    CharStringSubtype char;
+    NonCharStringSubtype nonChar;
+|};
+
+// allowed = false with empty `values` means `String:char`
```

Comment:
I know I used `boolean allowed;`, but coming back to the code I am finding it pretty confusing. Maybe `boolean valuesInOrOut;` then `const VALUES_IN = true; const VALUES_OUT = false;` ?

Reply:
how about `boolean modifier;` or `boolean adjective;`

with `const ALLOW = true;` `const DENY = false;` or `const REJECT = false;`

Reply:
The problem with these (as with `allowed`) is that the name applies to `values` but they are a field on the subtype.

Some other ideas:

```
   INCLUSIVE|EXCLUSIVE spec;
```

It also might be better to use +1 and -1 (as a boolean) to say respectively that the values are included or excluded.

Reply:
I tried `enum Spec { INCLUDE, EXCLUDE }`
It just bloated the downstream code. In my opinion, changing the name from `boolean allowed` to anything else didn't make it that much better either, do we need this change?